### PR TITLE
chore(cketh): add pagination to minter dashboard

### PR DESCRIPTION
**Pull request by lpahlavi**
([XC-240](https://dfinity.atlassian.net/browse/XC-240)) Add pagination to the ckETH minter dashboard.

[XC-240]: https://dfinity.atlassian.net/browse/XC-240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rs/ethereum/cketh/minter/src/dashboard/tests.rs
Comment (gregorydemay) [inline rs/ethereum/cketh/minter/src/dashboard/tests.rs:1101]: Generally: they are some minor issues regarding the pagination (FYI a failing test here we spit out the rendered html that you can then copy/paste in some `.html` file and open it in your web-browser to see the rendering). That's how I detected the following:

1. Spurious curly brace in the title
Screenshot 2024-12-12 at 11 01 11

The pagination does not seem to work properly: page 2 is not clickable and and transactions on page 1 start with 101.
```
<td colspan="8" style="text-align: right;">
        <b>Pages: </b>
        <a href="?finalized_transactions_start=0#finalized-transactions">1</a>&nbsp;
        2&nbsp;
        <a href="?finalized_transactions_start=200#finalized-transactions">3</a>&nbsp;
    </td>
```

Comment (lpahlavi) [reply rs/ethereum/cketh/minter/src/dashboard/tests.rs:1101]: I had a quick look but not sure yet where the extra curly brace is coming from. Thanks for the heads up, I would've missed that!

The page 2 link not working and the transactions starting at 101 are both expected though, since this test should show the second page of finalized transactions (i.e. transactions 101-200).

File: rs/ethereum/cketh/minter/src/dashboard.rs
Comment (gregorydemay) [inline rs/ethereum/cketh/minter/src/dashboard.rs:181]: nit: It's usually better to not restrict your type at this point and add any trait bound as needed only in the `impl` block.

File: rs/ethereum/cketh/minter/src/dashboard.rs
Comment (gregorydemay) [inline rs/ethereum/cketh/minter/src/dashboard.rs:195]: nit: if you need ownership, it's usually better to have the method signature saying so rather than deferencing/cloning -> `current_page_offset: usize`,

File: rs/ethereum/cketh/minter/src/dashboard/tests.rs
Comment (gregorydemay) [inline rs/ethereum/cketh/minter/src/dashboard/tests.rs:1028]: nit: maybe extract it to a method like `minted_event_with_log_index(index: &str) -> Vec<String>` so that it's a bit clearer that what matters here is the index

File: rs/ethereum/cketh/minter/src/dashboard/tests.rs
Comment (gregorydemay) [reply rs/ethereum/cketh/minter/src/dashboard/tests.rs:1128]: > The page 2 link not working and the transactions starting at 101 are both expected though, since this test should show the second page of finalized transactions (i.e. transactions 101-200).

You're right, I was a bit too fast when reading the test and I missed that.

**Review by gregorydemay — APPROVED** [review]
Thanks @lpahlavi for this PR. I have some minor comments but otherwise LGTM! Before merging, we should deploy it on the Sepolia minter to see if everything is as expected.

File: rs/ethereum/cketh/minter/src/dashboard.rs
Comment (lpahlavi) [reply rs/ethereum/cketh/minter/src/dashboard.rs:195]: This was actually a bit of a hack because this method was called in the HTML template and Askama passed a reference here for some reason.. But it's not needed anymore.

File: rs/ethereum/cketh/minter/templates/dashboard.html
Comment (lpahlavi) [reply rs/ethereum/cketh/minter/templates/dashboard.html:272]: So I think it makes sense to extract this function from the HTML template file, but then I thought I can just add the table directly to the dashboard struct. What do you think?

Comment (gregorydemay) [reply rs/ethereum/cketh/minter/templates/dashboard.html]: yes, good idea!

File: rs/ethereum/cketh/minter/templates/pagination.html
Comment (ninegua) [inline rs/ethereum/cketh/minter/templates/pagination.html:8]: This seems to discard the pagination data for other tables. So is it not possible to view both page 3 of one table and page 2 of another table?

Comment (lpahlavi) [reply rs/ethereum/cketh/minter/templates/pagination.html:8]: @ninegua Good question! You are correct. The problem is, afaik it is not trivial to add a query parameter without discarding the existing ones. It might be possible with a little bit of JS magic (see e.g. [here](https://stackoverflow.com/questions/6899097/how-to-add-a-parameter-to-the-url)), but I guess the question is whether it is worth the time. What do you think?

**Review by ninegua — APPROVED** [review]

**Pull request merged**
